src/files/files-list/FilesList.js

```diff
@@ -212,31 +196,39 @@ export class FilesList extends React.Component {
 
     if (e.key === 'ArrowDown' || e.key === 'ArrowUp') {
       e.preventDefault()
-      let index = this.props.upperDir ? -1 : 0
+      let index = upperDir ? -1 : 0
 
       if (focused !== null) {
-        const prev = this.props.files.findIndex(el => el.name === focused)
+        const prev = files.findIndex(el => el.name === focused)
         index = (e.key === 'ArrowDown') ? prev + 1 : prev - 1
       }
 
-      if (index === -1 && !this.props.upperDir) {
+      if (index === -1 && !upperDir) {
         return
       }
 
-      if (index >= -1 && index < this.props.files.length) {
+      if (index >= -1 && index < files.length) {
         let name
 
         if (index === -1) {
           name = '..'
         } else {
-          name = this.props.files[index].name
+          name = files[index].name
+        }
+
+        // If the file we are going to focus is out of view (removed
+        // from the DOM by react-virtualized), focus the first visible file
+        if (!this.filesRefs[name]) {
+          name = files[firstVisibleRow].name
```

Comment:
Perhaps one for a future PR, but I think we can

- use `scrollToCell` in react virtulized 
  - https://github.com/bvaughn/react-virtualized/blob/master/docs/Grid.md#scrolltocell--columnindex-number-rowindex-number-
- store `focused` as its array index not its file name.
- remove `this.fileRefs` entirely. It slowly fills up with a dom node for all items, which defeats some of the value of using react-virtualized.